File: 49. Group Anagrams/49. Group Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagramGroups = new HashMap<>();
        
        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sortedStr = new String(chars);
            
            anagramGroups.computeIfAbsent(sortedStr, k -> new ArrayList<>()).add(str);
        }
        
        return new ArrayList<>(anagramGroups.values());
    }
}

File: 49. Group Anagrams/49. Group Anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    anagram_groups = Hash.new { |h, k| h[k] = [] }
    
    strs.each do |str|
        sorted_str = str.chars.sort.join
        anagram_groups[sorted_str] << str
    end
    
    anagram_groups.values
end